Module.php

```diff
@@ -113,9 +113,13 @@ public function getContentContainerDescription(ContentContainerActiveRecord $con
     public function disableContentContainer(ContentContainerActiveRecord $container)
     {
         parent::disableContentContainer($container);
-        /** @var ExternalCalendar $item */
-        foreach (ExternalCalendar::find()->contentContainer($container)->each() as $item) {
-            $item->hardDelete();
+        /** @var ExternalCalendar $calendar */
+        foreach (ExternalCalendar::find()->contentContainer($container)->each() as $calendar) {
```

Comment:
@marc-farre   Ideally the `Externalcalendar::beforeDelete()` is removing all entries. Or isn't this triggered by the `hardDelete()`?

Reply:
This is what I thought too, but without removing the items before, the hard deletion of the calendar creates a timeout.

And for complete module disabling, items are deleted before: https://github.com/humhub-contrib/external_calendar/blob/master/Module.php#L64-L66

However, I think you are right.
What do you think about:
- removing https://github.com/humhub-contrib/external_calendar/blob/master/Module.php#L64-L66
- and here: https://github.com/humhub-contrib/external_calendar/blob/master/models/ExternalCalendar.php#L249
Replacing `ExternalCalendarEntry::findAll(['calendar_id' => $this->id])`
With `ExternalCalendarEntry::find()->where(['calendar_id' => $this->id])->each()`

Reply:
@marc-farre Your suggestion sounds good. Let's do it that way!

In the future it would of course be good to implement all “disables” as ActiveJob to avoid timeouts.

Reply:
@luke- Done in commit https://github.com/humhub-contrib/external_calendar/pull/76/commits/df6d5128ad7d1ef567807fc9709e284ba3cbd32a